src/emulator/auth/operations.ts

```diff
@@ -944,12 +941,23 @@ export function setAccountInfoImpl(
 
     if (reqBody.email) {
       assert(isValidEmailAddress(reqBody.email), "INVALID_EMAIL");
+      if (!emulatorUrl) {
+        throw new Error("Internal assertion error: Emulator URL invalid");
+      }
+
       const newEmail = canonicalizeEmailAddress(reqBody.email);
       if (newEmail !== user.email) {
         assert(!state.getUserByEmail(newEmail), "EMAIL_EXISTS");
         updates.email = newEmail;
         // TODO: Set verified if email is verified by IDP linked to account.
         updates.emailVerified = false;
+        if (user.email && !user.initialEmail) {
+          updates.initialEmail = user.email;
```

Comment:
BLOCKING: This is not the only place where an email can be added to an account. `signUp` and `signInWithEmailLink` and a few other places can too. It's probably better to do it in `state.ts` in `updateUserByLocalId` so that we can capture all cases.

Reply:
Agreed. But the prod behaviour is to set `initialEmail` only in RECOVER_EMAIL flow.

Reply:
Makes sense.